### PR TITLE
implement sceAppMgrLoadExec.

### PR DESCRIPTION
**Pull request by Zangetsu38**
- Implement sceAppMgrLoadExec

# Result: 
- Retroarch, god of war collection working 
- On Snes9x 2010
image

### CODE REVIEW

**Comment by Zangetsu38** [comment]
is ready for review

**Comment by Co2333** [comment]
It is better to do like this, otherwise our Xcode will run into error.

```
EXPORT(SceInt32, sceAppMgrLoadExec, const char *appPath, Ptr<char> const argv[], const SceAppMgrLoadExecOptParam *optParam) {
    if (optParam != nullptr)
        return SCE_APPMGR_ERROR_INVALID;
    
    SceSize _size;
    Ptr<void> _argv;
    
    if (!argv || !argv->get(host.mem)) {
        _size = SceSize(strlen(argv->get(host.mem)));
    } else {
        _size = 0;
    }

    if (argv != nullptr) {
        // Invoke values before -> Conditional expression is ambiguous; 'Ptr<void>' can be converted to 'const Ptr<char>' and vice versa
        _argv = *argv;
    }
    
    ExitCode ret = load_exec(host, static_cast<const std::string>(appPath), _size, _argv);
    
    if (ret == Success)
        return 0;

    return SCE_APPMGR_ERROR_INVALID_SELF_PATH;
}
```

**Comment by Zangetsu38** [comment]
@Co2333 is not xcode have get error, and iu have changed my code for get more smart, your version is nnot good, i prefere my first version ^^ 
but is fixed.

**Comment by Zangetsu38** [comment]
small rezworked with finnaly fix erase running thread with using swap for canswap on news thread before erase current, also using thread_id and no need create temp value on state

File: vita3k/modules/SceAppMgr/SceAppMgrUser.cpp [hunk 1]
```diff
@@ -328,8 +311,77 @@ EXPORT(int, sceAppMgrLaunchVideoStreamingApp) {
     return UNIMPLEMENTED();
 }
 
-EXPORT(int, sceAppMgrLoadExec) {
-    return UNIMPLEMENTED();
+EXPORT(SceInt32, sceAppMgrLoadExec, const char *appPath, Ptr<char> const argv[], const SceAppMgrLoadExecOptParam *optParam) {
+    if (optParam != nullptr)
```
[the remark this answers is not in the document]
Comment (Zangetsu38) [reply vita3k/modules/SceAppMgr/SceAppMgrUser.cpp]: yes, reserved area not used, and return it if detect value

File: vita3k/modules/SceAppMgr/SceAppMgrUser.cpp [hunk 2]
```diff
@@ -328,8 +311,77 @@ EXPORT(int, sceAppMgrLaunchVideoStreamingApp) {
...
+            if (argv && argv->get(host.mem)) {
+                while (argv[size_argv].get(host.mem))
+                    size_argv++;
+
+                if (size_argv > 1024)
+                    return RET_ERROR(SCE_APPMGR_ERROR_TOO_LONG_ARGV);
+            }
+
+            // Start exec thread
+            const auto exec = start_thread(host.kernel, exec_thread_id, size_argv, argv == nullptr ? Ptr<void>() : Ptr<void>(*argv));
```
[the remark this answers is not in the document]
Comment (Zangetsu38) [reply vita3k/modules/SceAppMgr/SceAppMgrUser.cpp]: i have not see any game using multiple argument for now, only one, and Ptr<void>() is obligatory for one travis lol

[the remark this answers is not in the document]
Comment (1whatleytay) [reply vita3k/modules/SceAppMgr/SceAppMgrUser.cpp]: Okay but it might cause problems... We should keep this in mind.

**Pull request merged**